yellowbrick/features/scatter.py

```diff
@@ -0,0 +1,278 @@
+            If classes is None and a y value is passed to fit then the classes
+            are selected from the target vector.
+
+        :param color: optional list or tuple of colors to colorize points
+            Use either color to colorize the points on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param colormap: optional string or matplotlib cmap to colorize points
+            Use either color to colorize the points on a per class basis or
+            colormap to color them on a continuous scale.
+
+        :param markers: iterable of strings
+            Matplotlib style markers for points on the scatter plot points
+
+        :param kwargs: keyword arguments passed to the super class.
+
+        These parameters can be influenced later on in the visualization
+        process, but can and should be set as early as possible.
+        """
+        super(ScatterVisualizer, self).__init__(ax, features, classes, color,
+                                                colormap, **kwargs)
+        self.markers = itertools.cycle(
+            kwargs.pop('markers', (',', '+', 'o', '*', 'v', 'h', 'd')))
+
+        # Ensure with init that features doesn't have more than two features
+        if features is not None:
+            if len(features) != 2:
+                raise YellowbrickValueError(
+                    'ScatterVisualizer only accepts two features')
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the parallel coords
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with 2 features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
+        """
+        _, ncols = X.shape
+
+        if ncols == 2:
+            X_two_cols = X
+            if self.features_ is None:
+                self.features_ = [0, 1]
+
+        # Handle the feature names if they're None.
+        elif self.features_ is not None and is_dataframe(X):
+            X_two_cols = X[self.features_].as_matrix()
+        else:
+            raise YellowbrickValueError("""
+                ScatterVisualizer only accepts two features, please
+                explicitly set these two features in the init kwargs or
+                pass a matrix/ dataframe in with only two columns.""")
+
+        # Store the classes for the legend if they're None.
+        if self.classes_ is None:
+            # TODO: Is this the most efficient method?
+            self.classes_ = [str(label) for label in set(y)]
```

Comment:
This is perfectly fine, but I wanted to point out `np.unique`, which I've been considering  using instead of the python `set` -- I don't know if it's faster or not, but if it is, it might be better. If you have any thoughts, I'd be happy to hear them.